docs/utilities/idempotency.md

```diff
@@ -0,0 +1,797 @@
+        payloadHash string
+    }
+    class Status {
+        <<Enumeration>>
+        INPROGRESS
+        COMPLETE
+        EXPIRED internal_only
+    }
+    IdempotencyRecord -- Status
+```
+
+<i>Idempotency record representation</i>
+</center>
+
+## Getting started
+
+### IAM Permissions
+
+Your Lambda function IAM Role must have `dynamodb:GetItem`, `dynamodb:PutItem`, `dynamodb:UpdateItem` and `dynamodb:DeleteItem` IAM permissions before using this feature.
+
+???+ note
+    If you're using one of our examples: [AWS Serverless Application Model (SAM)](#required-resources), [AWS Cloud Development Kit (CDK)](#required-resources), or [Terraform](#required-resources) the required permissions are already included.
+
+### Required resources
+
+Before getting started, you need to create a persistent storage layer where the idempotency utility can store its state - your lambda functions will need read and write access to it.
+
+As of now, Amazon DynamoDB is the only supported persistent storage layer, so you'll need to create a table first.
+
+**Default table configuration**
+
+If you're not [changing the default configuration for the DynamoDB persistence layer](#dynamodbpersistencelayer), this is the expected default configuration:
+
+| Configuration      | Value        | Notes                                                                               |
+| ------------------ | ------------ | ----------------------------------------------------------------------------------- |
+| Partition key      | `id`         |
+| TTL attribute name | `expiration` | This can only be configured after your table is created if you're using AWS Console |
+
+???+ tip "Tip: You can share a single state table for all functions"
+    You can reuse the same DynamoDB table to store idempotency state. We add the Lambda function name in addition to the idempotency key as a hash key.
+
+<!-- TODO: review CDK template -->
+=== "AWS Serverless Application Model (SAM) example"
+
+    ```yaml hl_lines="6-14 24-31"
+    Transform: AWS::Serverless-2016-10-31
+    Resources:
+    IdempotencyTable:
+        Type: AWS::DynamoDB::Table
+        Properties:
+        AttributeDefinitions:
+            - AttributeName: id
+            AttributeType: S
+        KeySchema:
+            - AttributeName: id
+            KeyType: HASH
+        TimeToLiveSpecification:
+            AttributeName: expiration
+            Enabled: true
+        BillingMode: PAY_PER_REQUEST
+
+    HelloWorldFunction:
+        Type: AWS::Serverless::Function
+        Properties:
+        Runtime: nodejs18.x
+        Handler: index.handler
+        Policies:
+            - Statement:
+            - Sid: AllowDynamodbReadWrite
+                Effect: Allow
+                Action:
+                - dynamodb:PutItem
+                - dynamodb:GetItem
+                - dynamodb:UpdateItem
+                - dynamodb:DeleteItem
+                Resource: !GetAtt IdempotencyTable.Arn
+    ```
+
```

Comment:
Good catch, I removed the reference from the sentence above for now.

I still plan on adding it but since the example in the Python docs was written in Python I need to write one in TS and I didn't have time to do it yet. I'll add the snippets in a future PR.

Reply:
I see, let's keep it like that for now